src/main/java/io/prestosql/plugin/tiledb/TileDBPageSink.java

```diff
@@ -415,52 +413,52 @@ else if (DATE.equals(type)) {
+                    value = type.getLong(block, position) * 1000000;
                     break;
                 }
                 case TILEDB_DATETIME_PS: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusNanos((long) ((type.getLong(block, position)) * 0.01)).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) * 100000;
                     break;
                 }
                 case TILEDB_DATETIME_NS: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusNanos((type.getLong(block, position))).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) * 10000;
                     break;
                 }
                 case TILEDB_DATETIME_US: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusNanos((type.getLong(block, position)) * 1000).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) * 1000;
                     break;
                 }
-                case TILEDB_DATETIME_DAY:
-                case TILEDB_DATETIME_MS: {
+                case TILEDB_DATETIME_MS:
+                case TILEDB_DATETIME_DAY: {
                     value = type.getLong(block, position);
                     break;
                 }
                 case TILEDB_DATETIME_SEC: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusSeconds((type.getLong(block, position))).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) / 1000;
                     break;
                 }
                 case TILEDB_DATETIME_MIN: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusMinutes((type.getLong(block, position))).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) / (60 * 1000);
                     break;
                 }
                 case TILEDB_DATETIME_HR: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusHours((type.getLong(block, position))).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) / (60 * 60 * 1000);
                     break;
                 }
                 case TILEDB_DATETIME_WEEK: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusWeeks((type.getLong(block, position))).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) / 7;
                     break;
                 }
                 case TILEDB_DATETIME_MONTH: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusMonths((type.getLong(block, position))).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) / 30;
```

Comment:
Does these conversions work for week/month/year? I think for these you will need to deal with actual timestamp objects inorder to account for variations in leap years and months right?

Reply:
That's correct. I'll do the required changes